### PR TITLE
Documentation fixes

### PR DESCRIPTION
**Pull request by froozen**
This fixes a few issues brought up in the `documentation` pull-request after it was closed as well as improving the quality of the documentation further.

### CODE REVIEW

File: src/Network/Ricochet/Connection.hs [hunk 1]
```diff
@@ -45,25 +53,26 @@ connectTo :: String              -- ^ Tor hidden service identifier to connect t
...
 initiateConnection :: Handle              -- ^ Handle corresponding to the connection to the peer
-                   -> Bool                -- ^ Wether this side accepted the connection
+                   -> ConnectionRole      -- ^ The connection role of this side of the connection
                    -> Ricochet Connection -- ^ Returns the finished 'Connection'
-initiateConnection handle isClientSide = do
-  connection <- createConnection handle
-  if isClientSide
+initiateConnection handle role = do
+  connection <- createConnection handle role
+  if role == Client
```
Comment (photm5) [inline src/Network/Ricochet/Connection.hs]: This would be better off as a case match.

File: src/Network/Ricochet/Monad.hs [hunk 2]
```diff
@@ -1,3 +1,11 @@
+{-|
+  Module:      Network.Ricochet.Monad
+  Description: Implementation of the 'Ricochet'-Monad
+
+"Network.Ricochet.Monad" contains the definition and implementation of the
+'Ricochet'-Monad, as well as some useful function in it.
```
Comment (photm5) [inline src/Network/Ricochet/Monad.hs]: Should be “functions.”

File: src/Network/Ricochet/Monad.hs [hunk 3]
```diff
@@ -37,11 +45,11 @@ newtype Ricochet a = Ricochet { runRicochet :: StateT RicochetState IO a }
...
 data RicochetState = MkRicochetState
-  { _serverSocket :: Socket
-  , _connections  :: [Connection]
-  , _contactList  :: [Contact]
-  , _socksPort    :: PortID
-  , _versions     :: Map Word8 (Connection -> Ricochet ())
+  { _serverSocket :: Socket                                -- ^ The socket listening for new peers
+  , _connections  :: [Connection]                          -- ^ A list of the current open connections
+  , _contactList  :: [Contact]                             -- ^ A list of known contacts
+  , _socksPort    :: PortID                                -- ^ The port of the local tor SOCKS proxy
```
Comment (photm5) [inline src/Network/Ricochet/Monad.hs]: Should be “Tor.”

File: src/Network/Ricochet/Protocol/Lowest.hs [hunk 4]
```diff
@@ -1,3 +1,11 @@
+{-|
+  Module:      Network.Ricochet.Protocol.Lowest
+  Description: Implementation of the low-level protocol parsing
+
+"Network.Ricochet.Protocol.Lowest" implements the parsing of the parts of the
+protocol, which aren't described via Google Protof.
```
Comment (photm5) [inline src/Network/Ricochet/Protocol/Lowest.hs]: Comma should be removed, and “Protof” should be “Protobuf.”

File: src/Network/Ricochet/Types.hs [hunk 5]
```diff
@@ -11,41 +19,45 @@ import           System.IO       (Handle ())
...
+-- | Creates a packet with appropriate size from a Channel and a ByteString
 makePacket :: Word16     -- ^ ID of the channel the packet should be sent on
            -> ByteString -- ^ The ByteString to be sent
            -> Packet     -- ^ Returns a sendable packet
 makePacket chan bs = MkPacket (4 + fromIntegral (B.length bs)) chan bs
 
+-- | The role of a peer in a Connection
+data ConnectionRole = Client | Server deriving (Eq, Show)
+
 -- | Representation of a connection between two ricochet users
```
Comment (photm5) [inline src/Network/Ricochet/Types.hs]: “Users” should be “peers”, but I’m not sure whether it belongs in this PR.

File: src/Network/Ricochet/Types.hs [hunk 6]
```diff
@@ -11,41 +19,45 @@ import           System.IO       (Handle ())
...
 -- it consists of:
 --  - open channels
 --  - wether our ricochet instance is the client
 -- Equality is defined by equality of the socket
```
Comment (photm5) [inline src/Network/Ricochet/Types.hs]: I still think this should be moved to the instance definition.  Did you try out if it works?

File: src/Network/Ricochet/Types.hs [hunk 7]
```diff
@@ -11,41 +19,45 @@ import           System.IO       (Handle ())
...
+  in  MkConnection handle channels B.empty role
 
 instance Eq Connection where
   a == b = _cHandle a == _cHandle b
 
 -- | Low level representation of a ricochet channel
 data Channel = MkChannel
-  { _cChannelID   :: Word16
-  , _cChannelType :: ChannelType
+  { _cChannelID   :: Word16      -- ^ The channels ID
```
Comment (photm5) [inline src/Network/Ricochet/Types.hs]: Should be “channel’s.” I’m not sure whether this should be changed to resemble the line below though.

File: src/Network/Ricochet/Types.hs [hunk 8]
```diff
@@ -53,8 +65,8 @@ data ChannelType = MkChannelType ByteString
 
 -- | A contact of our user, defined by his ID (TOR hidden service address without the .onion and the 'ricochet:' prefix) and his display name
```
Comment (photm5) [inline src/Network/Ricochet/Types.hs]: “Our user” sounds as if we would store information about the user that uses the client that is developed using this library.

File: src/Network/Ricochet/Types.hs (hunk 6 above)
Comment (froozen) [inline src/Network/Ricochet/Types.hs]: Just tried it and it actualy works.

File: src/Network/Ricochet/Types.hs (hunk 5 above)
Comment (sternenseemann) [inline src/Network/Ricochet/Types.hs]: That should be fixed in this PR.

**Pull request merged**